compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -239,7 +152,7 @@ FailureOr<PadPackTiling> PadPackTiling::create(linalg::LinalgOp linalgOp) {
   const uint64_t K = lhsShape[1];
 
   FailureOr<unsigned> maybeScaleFactor =
-      getTilingScaleFactor(lhsType.getElementType());
+      getTilingScaleFactor(initType.getElementType());
```

Comment:
I guess 'scale factor' isn't really the right abstraction for this. For an i8 -> i32 matmul, I don't think it makes sense to use the (small) scale factor for i32 to determine how many i8 elements can fit in AIE core memory or memtile memory. We should revisit this later..

Reply:
I'm not sure which factor is correct to use. My thinking is we need to initialize and do computation on the output buffer in the AIE core, so we need to take into account the larger data type.